Dungeons/Dungeons/CorridorNodeLayouter.cs

```diff
@@ -96,39 +96,59 @@ protected virtual void LayoutNodes(DungeonNode level, List<DungeonNode> mazeNode
       var d = new DungeonGenerator(container);
       var infoC = new GenerationInfo();
       infoC.NumberOfRooms = 1;
+      var numberOfNodes = mazeNodes.Count;
 
       mazeNodes[0].Placement = RoomPlacement.LeftUpper;
-      mazeNodes[1].Placement = RoomPlacement.RightUpper;
-      mazeNodes[2].Placement = RoomPlacement.RightLower;
-      mazeNodes[3].Placement = RoomPlacement.LeftLower;
-      mazeNodes[4].Placement = RoomPlacement.Center;
-
       mazeNodes[0].GenerateDoors(RoomPlacement.LeftUpper);
-      mazeNodes[1].GenerateDoors(RoomPlacement.LeftLower);
-      mazeNodes[2].GenerateDoors(RoomPlacement.RightUpper);
-      mazeNodes[3].GenerateDoors(RoomPlacement.RightLower);
-      mazeNodes[4].GenerateDoors(RoomPlacement.Center);
+      if (mazeNodes.Count > 1)
+      {
+        mazeNodes[1].Placement = RoomPlacement.RightUpper;
+        mazeNodes[1].GenerateDoors(RoomPlacement.LeftLower);
+      }
+      if (mazeNodes.Count > 2)
+      {
+        mazeNodes[2].Placement = RoomPlacement.RightLower;
+        mazeNodes[2].GenerateDoors(RoomPlacement.RightUpper);
+      }
+      if (mazeNodes.Count > 3)
+      {
+        mazeNodes[3].Placement = RoomPlacement.LeftLower;
+        mazeNodes[3].GenerateDoors(RoomPlacement.RightLower);
+      }
+      if (mazeNodes.Count > 4)
+      {
+        mazeNodes[4].Placement = RoomPlacement.Center;
+        mazeNodes[4].GenerateDoors(RoomPlacement.Center);
+      }
 
       var conn = new DungeonNodeConnector(); //generating normal rooms
-      for (int i = 0; i < 4; i++)
+      level.AppendMaze(mazeNodes[0], new Point(0, 0));
```

Comment:
Tu by tez fanie bylo urpościć, chyba coś w tym stylu:
var conn = new DungeonNodeConnector(); //generating normal rooms
      var nodesCopy = mazeNodes.ToList();
      for (int i = 0; i < nodesCopy.Count-1; i++)
      {
        var corridor = conn.ConnectNodes(nodesCopy, nodesCopy[i], nodesCopy[i + 1], d);
        mazeNodes.Add(corridor);
        level.AppendMaze(nodesCopy[i], conn.Node1Position);
        level.AppendMaze(nodesCopy[i+1], conn.Node2Position);
        level.AppendMaze(corridor, conn.CorridorPosition);
      }